ui/components/src/Legend/ListLegend.tsx

```diff
@@ -11,28 +11,110 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { List } from '@mui/material';
+import { useTheme } from '@mui/material';
+import { VariableSizeList, ListChildComponentProps } from 'react-window';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { useRef, useEffect, forwardRef } from 'react';
 import { LegendItem } from '../model';
 import { ListLegendItem } from './ListLegendItem';
 
 interface ListLegendProps {
   items: LegendItem[];
 }
 
```

Comment:
This code is a little gnarlier than I would like because we need to:
- Virtualize
- Simulate padding -- if you use real padding/margin, the scrollbar ends up offset too, which looks yucky.
- Handle dynamic heights because the items change on hover

The implementation is based on guidance from react-window, so I think it's a reasonable approach, but not super-easy to follow, so I added a lot of comments to try to help understand.